### PR TITLE
Reorganize distributed tree files

### PR DESCRIPTION
**Pull request by aprokop**
I had a lot of trouble navigating `ArborX_DetailsDistributedTreeImpl.hpp` file while working on APIv2 change, and understanding how parts fit together, and what is used by the nearest and by the spatial search.

This PR splits `ArborX_DetailsDistributedTreeImpl.hpp` into several files. There is almost no new code, just things moving around. One minor change is making `DistributedTreeImpl` non-templated, and moving the utility functions out of the struct.

New files:
- `ArborX_DetailsDistributedTreeSpatial.hpp` containing the spatial `queryDispatch`
- `ArborX_DetailsDistributedTreeNearest.hpp` containing the nearest `queryDispatch`
- `ArborX_DetailsDistributedTreeUtils.hpp` containing the helper routines (`forwardQueries`, `communicateResultsBack`, etc)
- `ArborX_DetailsDistributedTreeImpl.hpp` containing only the friend struct `DistributedTreeImpl` to be a friend of `DistributedTree`

### CODE REVIEW

**Comment by masterleinad** [comment]
> One minor change is making DistributedTreeImpl non-templated, and moving the utility functions out of the struct.

Make it a separate commit/pull request.

**Comment by aprokop** [comment]
> Make it a separate commit/pull request.

No. It would require me to almost redo the whole thing for little gain.

**Comment by aprokop** [comment]
Rebased on `master` to fix conflicts.

File: src/ArborX_DistributedTree.hpp [hunk 1]
```diff
@@ -28,11 +28,6 @@
 namespace ArborX
 {
 
-/** \brief Distributed search tree
- *
- *  \note query() must be called as collective over all processes in the
- *  communicator passed to the constructor.
```
Comment (dalg24) [inline src/ArborX_DistributedTree.hpp:34]: Is that note captured elsewhere?

**Review by dalg24 — COMMENTED** [review]
ok with the general direction

File: src/ArborX_DistributedTree.hpp [hunk 2]
```diff
@@ -28,11 +28,8 @@
...
- */
+// NOTE: query() must be called as collective over all processes in the
+// communicator passed to the constructor
```
Comment (dalg24) [inline src/ArborX_DistributedTree.hpp:32]: I think you should withdraw all changes to the comment but I am not blocking

**Review by dalg24 — APPROVED** [review]

**Comment by aprokop** [comment]
Had to rebase on #1005 to fix conflicts.

**Pull request merged**